packages/web/src/components/collections-table/CollectionsTable.tsx

```diff
@@ -0,0 +1,263 @@
+
+type RowInfo = UserCollectionMetadata & {
+  name: string
+  artist: string
+  date: string
+  dateSaved: string
+  dateAdded: string
+}
+
+type CollectionCell = Cell<RowInfo>
+
+type CollectionRow = Row<RowInfo>
+
+export type CollectionsTableColumn =
+  | 'name'
+  | 'overflowMenu'
+  | 'releaseDate'
+  | 'reposts'
+  | 'saves'
+  | 'spacer'
+
+type CollectionsTableProps = {
+  columns?: CollectionsTableColumn[]
+  data: any[]
+  isVirtualized?: boolean
+  loading?: boolean
+  defaultSorter?: (a: any, b: any) => number
+  fetchMore?: (offset: number, limit: number) => void
+  fetchBatchSize?: number
+  onClickRow?: (collection: CollectionMetadata, index: number) => void
+  scrollRef?: React.MutableRefObject<HTMLDivElement | undefined>
+  showMoreLimit?: number
+  totalRowCount?: number
+}
+
+const defaultColumns: CollectionsTableColumn[] = [
+  'name',
+  'overflowMenu',
+  'releaseDate',
+  'reposts',
+  'saves',
+  'spacer'
+]
+
+/**
+ * Note: this table is not generalized the way TracksTable is. It is
+ * specifically tailored to the Artist Dashboard use case.
+ */
+export const CollectionsTable = ({
+  columns = defaultColumns,
+  data,
+  defaultSorter,
+  fetchBatchSize,
+  fetchMore,
+  isVirtualized = false,
+  loading = false,
+  onClickRow,
+  scrollRef,
+  showMoreLimit,
+  totalRowCount
+}: CollectionsTableProps) => {
+  // Cell Render Functions
+  const renderNameCell = useCallback((cellInfo: CollectionCell) => {
+    const collection = cellInfo.row.original
+    const deleted = collection.is_delete || !!collection.user?.is_deactivated
+
+    return (
+      <Flex
+        gap='xs'
+        css={{
+          overflow: 'hidden',
+          position: 'relative',
+          display: 'inline-flex',
+          'max-width': '100%'
+        }}
+      >
+        <TextLink
+          tag={deleted ? 'span' : undefined}
+          to={deleted ? '' : collection.permalink ?? ''}
+          textVariant='title'
+          size='s'
+          strength='weak'
+          css={{ display: 'block' }}
+          ellipses
+        >
+          {collection.playlist_name}
+          {deleted ? ` [Deleted By Artist]` : ''}
+        </TextLink>
+      </Flex>
+    )
+  }, [])
+
+  const renderRepostsCell = useCallback((cellInfo: CollectionCell) => {
+    const collection = cellInfo.row.original
+    return formatCount(collection.repost_count)
+  }, [])
+
+  const renderSavesCell = useCallback((cellInfo: CollectionCell) => {
+    const collection = cellInfo.row.original
+    return collection.save_count
+  }, [])
+
+  const renderReleaseDateCell = useCallback((cellInfo: CollectionCell) => {
+    const collection = cellInfo.row.original
+    let suffix = ''
+    if (
+      collection.release_date &&
+      moment(collection.release_date).isAfter(moment.now())
+    ) {
+      suffix = ' (Scheduled)'
+    }
+    return (
+      moment(collection.release_date ?? collection.created_at).format(
+        'M/D/YY'
+      ) + suffix
+    )
+  }, [])
```

Comment:
Feels like this logic isn't unique to this component. Do we have a format date or format release date helper?

Nbd if it's only used here and the tracks table though

Reply:
yep it's only here and tracks table.